src/Symfony/Component/AssetMapper/ImportMap/ImportMapGenerator.php

```diff
@@ -225,17 +230,20 @@ private function findEagerImports(MappedAsset $asset): array
                 continue;
             }
 
-            $dependencies[] = $javaScriptImport->importName;
+            $dependencies[$javaScriptImport->importName] = $javaScriptImport->importName;
 
             // Follow its imports!
             if (!$nextAsset = $this->assetMapper->getAsset($javaScriptImport->assetLogicalPath)) {
                 // should not happen at this point, unless something added a bogus JavaScriptImport to this asset
                 throw new LogicException(sprintf('Cannot find imported JavaScript asset "%s" in asset mapper.', $javaScriptImport->assetLogicalPath));
             }
-            $dependencies = array_merge($dependencies, $this->findEagerImports($nextAsset));
+
+            if ($nextImports = $this->findEagerImports($nextAsset)) {
+                $dependencies += array_combine($nextImports, $nextImports);
+            }
         }
 
-        return $dependencies;
+        return array_values($dependencies);
```

Comment:
```suggestion
        return array_unique($dependencies);
```

And removing the other changes?

Reply:
You're right... i often try to find the "smartest" way of doing this kind of things.... but it often end with a harder code lisibility and... no gain at all 

(🦦 i'm sure i won 0.001s in the sorting of my 200 items)

Hard habits :) Fixing asap